### DISCUSSION TITLE
getServerSideProps can not be exported error in Nextjs

### DISCUSSION THREAD

**Discussion by krishnahari6622** [vercel/next.js#42584] [Help] [answered]
Trying to create OUT file in next js, getting an error saying pages with `getServerSideProps` can not be exported.

**Comment by arielvieira** [comment]
@krishnahari6622 

what version of next are you using?
what you trying to export from a file inside the `pages` directory or a `app` directory?

**Reply by krishnahari6622** [reply]
@arielvieira I am using "next : 12.3.1" version. My project have a page called **Work** in which caseStudies must be shown to the user. So I am using getserversideprops to fetch the case studies through API. Things are working as expected in dev but when I tried to deploying the code to prod got the above error and work page(html ) is not created in OUT file as it is not a static page. Any idea or alternative how to deploy the serversideprops to deploy to prod?. Thank You.

**Reply by icyJoseph** [reply]
@krishnahari6622 refer to my answer

**Reply by krishnahari6622** [reply]
Hi @icyJoseph  I am new to Next js , can you plz suggest other ways to deploy the getserversideprops to prod.

**Reply by icyJoseph** [reply]
Look at my answer :) @krishnahari6622 edited with a few more tidbits.

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Are you using `next export`? `getServerSideProps` requires a Node.js runtime server to take care of running the function. When you do `next export` you ask Next.js to make a static HTML page, which, to put it simply, has no runtime server. 

`next export` generated applications require a static files server to be delivered to the user. 

You have to change your strategy when it comes to `next export`.

Here's a list, https://nextjs.org/docs/advanced-features/static-html-export#unsupported-features, of unsupported features, at the bottom of the list you can see `getServerSideProps`.

## Edit

Additionally, API routes do not work when you do `next export`. 

You'll really need to consider using `next build` and `next start` and that of course means having a runtime server. Use Vercel which has very generous free-tiers.

**Reply by icyJoseph** [reply]
@krishnahari6622 

If you just need to show your project from your local machine:

- `next build` (via `npm run build`) for example
- `next start` (via `npm run start`) 

The above builds the project and starts a server that'll handle incoming requests and in turn run `getServerSideProps`, API routes, and other features.

## If you want to share it with the world

You can create a GitHub repository with your code, and keep it private. Then go to `vercel.com` login with your GitHub account and link the repository you made to a Vercel project, https://vercel.com/new - follow the steps there and you'll be able to host your Next.js app with Vercel, which has fairly good limits. Read these:

- https://vercel.com/docs/concepts/limits/fair-use-policy
- https://vercel.com/docs/concepts/limits/overview

**Reply by krishnahari6622** [reply]
Got it. Thank you so much @icyJoseph

**Reply by icyJoseph** [reply]
Let us know how it goes. And when it works don't forget to mark as answered :)

**Reply by krishnahari6622** [reply]
@icyJoseph @arielvieira  linked the repo to vercel. Its working now . Thank You.

<img width="1410" alt="Screenshot 2022-11-07 at 10 26 24 PM" src="https://user-images.githubusercontent.com/54630482/200369794-10f9083d-dff4-4a52-9b59-368bb3d70782.png">

**Reply by icyJoseph** [reply]
Great! Please don't forget to mark as answered! Good luck!
